src/lib.rs.in

```diff
@@ -47,87 +46,103 @@ fn json_response(obj: &Value) -> IronResult<Response> {
 }
 
 
-/// Configuration data for a "famous" identity provider.
-///
-/// Used as part of the `AppConfig` struct.
+/// Holds runtime configuration data for this daemon instance.
 #[derive(Clone)]
-struct ProviderConfig {
-    /// URL pointing to the OpenID configuration document.
+pub struct AppConfig {
+    base_url: String, // Origin of this instance, used for constructing URLs
+    priv_key: PKey, // Signing key
+    store: redis::Client, // Redis Client
+    expire_keys: usize, // Redis key TTL, in seconds
+    sender: Email, // From address for email
+    token_validity: usize, // JWT validity duration, in seconds
+    providers: HashMap<String, Provider>, // Mapping of Domain -> OIDC Provider
+}
+
+
+/// Represents an OpenID Connect provider.
+#[derive(Clone, Deserialize)]
+pub struct Provider {
     discovery: String,
-    /// Client ID issued for this daemon instance by the provider.
     client_id: String,
-    /// Secret issued for this daemon instance by the provider.
     secret: String,
-    /// Issuer origin as used in identity tokens issued by the provider.
-    /// Used to check that the issuer in the token matches expectations.
-    issuer: String,
```

Comment:
I've added `issuer` to the test configuration file in b92ae45205. Can you change this so it works again?